### ISSUE TITLE
Double requests for documentHighlight

### ISSUE THREAD

**Issue by rchl** [sublimelsp/LSP#1108]
I've noticed that setting a cursor on some symbol makes the `documentHighlight` underline blink once.

Checked logs and saw the request being made twice on each cursor movement:
```
:: --> pyls textDocument/documentHighlight(12): {'textDocument': {'uri': 'file:////LSP/plugin/highlights.py'}, 'position': {'character': 8, 'line': 38}}
:: --> pyls textDocument/documentHighlight(13): {'textDocument': {'uri': 'file:////LSP/plugin/highlights.py'}, 'position': {'character': 8, 'line': 38}}
```

Then added log in `DocumentHighlightListener` class, inside `on_selection_modified_async` method and that listener seems to be triggered twice on cursor movement. Tested with `print(self.view.file_name())`.

**Comment by rwols** [comment]
on_selection_modified_async is invoked both on mousebutton down and on mousebutton up it seems. But not always.

https://github.com/sublimehq/sublime_text/issues/1254

**Comment by rwols** [comment]
Fixed by #1110
